components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -1358,6 +1358,76 @@ impl Time {
     pub fn invalid_zero(self) -> bool {
         self.month() == 0 || self.day() == 0
     }
+
+    pub fn from_days(ctx: &mut EvalContext, daynr: u32) -> Result<Self> {
+        let (year, month, day) = Time::get_date_from_daynr(daynr);
+        let time_args = TimeArgs {
+            year,
+            month,
+            day,
+            hour: 0,
+            minute: 0,
+            second: 0,
+            micro: 0,
+            fsp: 0,
+            time_type: TimeType::Date,
+        };
+        Time::new(ctx, time_args)
+    }
+
+    // Changes a daynr to year, month and day, daynr 0 is returned as date 00.00.00
+    #[inline]
+    fn get_date_from_daynr(daynr: u32) -> (u32, u32, u32) {
+        if daynr <= 365 || daynr >= 3_652_500 {
```

Comment:
```sql
mysql iosmanthus@(none):test> select from_days(3652424)                                                                           
+--------------------+
| from_days(3652424) |
+--------------------+
| 9999-12-31         |
+--------------------+
1 row in set
Time: 0.019s
mysql iosmanthus@(none):test> select from_days(3652425)                                                                           
+--------------------+
| from_days(3652425) |
+--------------------+
| 0000-00-00         |
+--------------------+
1 row in set
Time: 0.021s
```

Reply:
The implementation is copied from TiDB's, and MySQL 5.7's source code writes like this also:
```
// sql_time.cc

void get_date_from_daynr(long daynr, uint *ret_year, uint *ret_month, uint *ret_day)
{
...
    if (daynr <= 365L || daynr >= 3652500)
    {
        ...
    }
...
}
```
 Why isn't it exactly 3652424L, and it also confused me.